app/models/ReadsEstablisherDetails.scala

```diff
@@ -23,8 +23,9 @@ import scala.annotation.tailrec
 
 object ReadsEstablisherDetails {
 
-  private def previousAddressDetails(addressYears: String, previousAddress: Option[Address]): Option[PreviousAddressDetails] = {
-    if (addressYears == "under_a_year") {
+  def previousAddressDetails(addressYears: String, previousAddress: Option[Address],
+                                     tradingTime: Option[Boolean] = None): Option[PreviousAddressDetails] = {
+    if (addressYears == "under_a_year" && tradingTime!=Some(false)) {
```

Comment:
Replace with tradingTime.getOrElse(true)  - gets rid of compiler warning and is more readable